parkhangs-webapp/src/features/modal/ModalMapDetail.js

```diff
@@ -1,14 +1,23 @@
 import React from 'react'
 import AddEventForm from 'components/AddEventForm'
-import { deleteEvent } from 'features/parks/parksSlice.js'
+import { deleteEvent,fetchEventsById } from 'features/parks/parksSlice.js'
 import { connect } from 'react-redux'
 import 'features/modal/ModalMapDetail.css'
 
+
 class ModalMapDetail extends React.Component {
+ 
     selectedPark = () => {
         return this.props.parks.find(park => park.parkName === this.props.park)
     }
-
+   
+   componentDidMount(){
+    //currently promise is being rejected: figure out how to call the promise here and load into events.map area
+       console.log(this.props.fetchEvents(this.selectedPark().id))
```

Comment:
all you have to do is await this since it's async. I did this and it works. 
```
    componentDidMount(){
        this.getEvents()
    }

    getEvents = async () => {
       const events = await this.props.fetchEvents(this.selectedPark().id)
       console.log(events)
    }
```

Reply:
Also to resolve the promise and actually get back the payload from `fetchEvents()`, use the `unwrapResult` function from redux toolkit, like 
```
const resultAction = await this.props.fetchEvents(this.selectedPark().id)
       const events = unwrapResult(resultAction)
       console.log(events)
```

Reply:
thanks for point out async! I tried adding it into the mapToProps instead and that didn't work, the helper function makes sense. Thanks again Claire!